stl/inc/memory

```diff
@@ -1504,6 +1556,238 @@ private:
     }
 };
 
+#ifdef __cpp_concepts
+template <class _Ty>
+constexpr _Ty _Div_ceil(_Ty _Val1, _Ty _Val2) noexcept {
+    return (_Val1 % _Val2 == 0) ? (_Val1 / _Val2) : (_Val1 / _Val2 + 1);
+}

```

Comment:
Does this compute the division twice? If so one could use std::div and check the remainder. At least for integral types this should be fine.
```suggestion
template <class _Ty>
constexpr _Ty _Div_ceil(_Ty _Val1, _Ty _Val2) noexcept {
    const auto res = _STD div(_Val1, _Val2);
    return (res.rem == 0) ? (res.quot) : (res.quot + 1);
}
```

Reply:
I think compiler should be able to optimize it anyway but it's actually good to ensure it.